src/test/ui/mismatched_types/abridged.stderr

```diff
@@ -0,0 +1,63 @@
+error[E0308]: mismatched types
+  --> $DIR/abridged.rs:26:5
+   |
+26 |     Some(Foo { bar: 1 })
+   |     ^^^^^^^^^^^^^^^^^^^^ expected struct `Foo`, found enum `std::option::Option`
+   |
+   = note: expected type `Foo`
+              found type `std::option::Option<Foo>`
+   = help: here are some functions which might fulfill your needs:
+           - .unwrap()
+
+error[E0308]: mismatched types
+  --> $DIR/abridged.rs:30:5
+   |
+30 |     Foo { bar: 1 }
+   |     ^^^^^^^^^^^^^^ expected enum `std::option::Option`, found struct `Foo`
+   |
+   = note: expected type `std::option::Option<Foo>`
```

Comment:
Yeah... I've been thinking that we could expand ui errors to have a mode where ANSI escape codes are kept (`.stderr_full`? A comment flag in the `.rs` test file?).